examples/random-login/main.go

```diff
@@ -40,22 +40,21 @@ func provider() p.Provider {
 
 // TODO: Deserialization does not yet work for external resources. Right now, it looks
 // like this structure is only implementable in typescript, but that will need to change.
-type MoreRandomPassword struct{}
 type MoreRandomPasswordArgs struct {
 	Length *random.RandomInteger `pulumi:"length" provider:"type=random@v4.8.1:index/randomInteger:RandomInteger"`
 }
 
-type MoreRandomPasswordState struct {
+type MoreRandomPassword struct {
 	pulumi.ResourceState
 	Length   *random.RandomInteger  `pulumi:"length" provider:"type=random@v4.8.1:index/randomInteger:RandomInteger"`
 	Password *random.RandomPassword `pulumi:"password" provider:"type=random@v4.8.1:index/randomPassword:RandomPassword"`
 }
 
-func (r *MoreRandomPassword) Construct(ctx *pulumi.Context, name, typ string, args MoreRandomPasswordArgs, opts pulumi.ResourceOption) (*MoreRandomPasswordState, error) {
-	comp := &MoreRandomPasswordState{
+func NewMoreRandomPassword(ctx *pulumi.Context, name, typ string, args MoreRandomPasswordArgs, opts ...pulumi.ResourceOption) (*MoreRandomPassword, error) {
```

Comment:
I'm generally supportive of this change, but two questions:
1. Regular component "constructor functions" don't take a `typ string` arg. The type is specified inside the implementation of the function. I wonder if there's a way to avoid that in the signature if the function, and pass in the inferred type in some other way. Maybe through the ctx?
2. The order of the `typ` and `name` args in `ctx.RegisterComponentResource` are in the reverse order, with typ being first and name bing second. If we must take in `typ`, should we match the order of `ctx.RegisterComponentResource`?

Reply:
We have moved the type string into the local context.